Failure in `Installer Build and Test coreclr windows_x64 Release failed` - tests passed but writing to log failed:
```
  Running tests: D:\a\_work\1\s\artifacts\bin\Microsoft.NET.HostModel.AppHost.Tests\Release\net7.0\Microsoft.NET.HostModel.AppHost.Tests.dll [net7.0|x64]
  Running tests: D:\a\_work\1\s\artifacts\bin\Microsoft.NET.HostModel.ComHost.Tests\Release\net7.0\Microsoft.NET.HostModel.ComHost.Tests.dll [net7.0|x64]
  Running tests: D:\a\_work\1\s\artifacts\bin\AppHost.Bundle.Tests\Release\net7.0\AppHost.Bundle.Tests.dll [net7.0|x64]
  Tests succeeded: D:\a\_work\1\s\artifacts\bin\Microsoft.NET.HostModel.AppHost.Tests\Release\net7.0\Microsoft.NET.HostModel.AppHost.Tests.dll [net7.0|x64]
D:\a\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\7.0.0-beta.22266.1\tools\VSTest.targets(55,5): error MSB3491: Could not write lines to file "D:\a\_work\1\s\artifacts\log\Release\Microsoft.NET.HostModel.ComHost.Tests_net7.0_x64.log". The process cannot access the file 'D:\a\_work\1\s\artifacts\log\Release\Microsoft.NET.HostModel.ComHost.Tests_net7.0_x64.log' because it is being used by another process. [D:\a\_work\1\s\src\installer\tests\Microsoft.NET.HostModel.Tests\Microsoft.NET.HostModel.ComHost.Tests\Microsoft.NET.HostModel.ComHost.Tests.csproj]
##[error].packages\microsoft.dotnet.arcade.sdk\7.0.0-beta.22266.1\tools\VSTest.targets(55,5): error MSB3491: (NETCORE_ENGINEERING_TELEMETRY=Build) Could not write lines to file "D:\a\_work\1\s\artifacts\log\Release\Microsoft.NET.HostModel.ComHost.Tests_net7.0_x64.log". The process cannot access the file 'D:\a\_work\1\s\artifacts\log\Release\Microsoft.NET.HostModel.ComHost.Tests_net7.0_x64.log' because it is being used by another process.
  Running tests: D:\a\_work\1\s\artifacts\bin\Microsoft.NET.HostModel.Bundle.Tests\Release\net7.0\Microsoft.NET.HostModel.Bundle.Tests.dll [net7.0|x64]
  Tests succeeded: D:\a\_work\1\s\artifacts\bin\Microsoft.NET.HostModel.Bundle.Tests\Release\net7.0\Microsoft.NET.HostModel.Bundle.Tests.dll [net7.0|x64]
  Running tests: D:\a\_work\1\s\artifacts\bin\HostActivation.Tests\Release\net7.0\HostActivation.Tests.dll [net7.0|x64]
  Tests succeeded: D:\a\_work\1\s\artifacts\bin\AppHost.Bundle.Tests\Release\net7.0\AppHost.Bundle.Tests.dll [net7.0|x64]
  Tests succeeded: D:\a\_work\1\s\artifacts\bin\HostActivation.Tests\Release\net7.0\HostActivation.Tests.dll [net7.0|x64]
```

@dotnet/jit-contrib

--- a/src/coreclr/jit/lclmorph.cpp
+++ b/src/coreclr/jit/lclmorph.cpp
@@ -1055,11 +1055,10 @@ private:
             return IndirTransform::None;
         }
 
-        if ((user == nullptr) || !user->OperIs(GT_ASG))
+        if ((user == nullptr) || !user->OperIs(GT_ASG, GT_RETURN))
         {
-            // TODO-ADDR: Skip TYP_STRUCT indirs for now, unless they're used by an ASG.
-            // At least call args will require extra work because currently they must be
-            // wrapped in OBJ nodes so we can't replace those with local nodes.
+            // TODO-ADDR: call args require extra work because currently they must
+            // be wrapped in OBJ nodes so we can't replace those with local nodes.
             return IndirTransform::None;
         }
 
@@ -1105,26 +1104,26 @@ private:
 
         // Current matrix of matches/users/types:
         //
-        // |------------|------|---------|--------|
-        // | STRUCT     | CALL | ASG     | RETURN |
-        // |------------|------|---------|--------|
-        // | Exact      | None | LCL_VAR | None   |
-        // | Compatible | None | LCL_VAR | None   |
-        // | Partial    | None | OBJ     | None   |
-        // |------------|------|---------|--------|
+        // |------------|------|---------|---------|
+        // | STRUCT     | CALL | ASG     | RETURN  |
+        // |------------|------|---------|---------|
+        // | Exact      | None | LCL_VAR | LCL_VAR |
+        // | Compatible | None | LCL_VAR | LCL_VAR |
+        // | Partial    | None | OBJ     | LCL_FLD |
+        // |------------|------|---------|---------|
         //
-        // |------------|------|---------|--------|----------|
-        // | SIMD       | CALL | ASG     | RETURN | HWI/SIMD |
-        // |------------|------|---------|--------|----------|
-        // | Exact      | None | None    | None   | None     |
-        // | Compatible | None | None    | None   | None     |
-        // | Partial    | None | None    | None   | None     |
-        // |------------|------|---------|--------|----------|
+        // |------------|------|---------|---------|----------|
+        // | SIMD       | CALL | ASG     | RETURN  | HWI/SIMD |
+        // |------------|------|---------|---------|----------|
+        // | Exact      | None | None    | None    | None     |
+        // | Compatible | None | None    | None    | None     |
+        // | Partial    | None | None    | None    | None     |
+        // |------------|------|---------|---------|----------|
         //
         // TODO-ADDR: delete all the "None" entries and always
         // transform local nodes into LCL_VAR or LCL_FLD.
 
-        assert(user->OperIs(GT_ASG) && indir->TypeIs(TYP_STRUCT));
+        assert(indir->TypeIs(TYP_STRUCT) && user->OperIs(GT_ASG, GT_RETURN));
 
         *pStructLayout = indirLayout;
 
@@ -1133,7 +1132,12 @@ private:
             return IndirTransform::LclVar;
         }
 
-        return IndirTransform::ObjAddrLclFld;
+        if (user->OperIs(GT_ASG))
+        {
+            return IndirTransform::ObjAddrLclFld;
+        }
+
+        return IndirTransform::LclFld;
     }
 
     //------------------------------------------------------------------------

--- a/src/coreclr/jit/lower.cpp
+++ b/src/coreclr/jit/lower.cpp
@@ -3686,6 +3686,10 @@ void Lowering::LowerRetStruct(GenTreeUnOp* ret)
             LowerRetSingleRegStructLclVar(ret);
             break;
 
+        case GT_LCL_FLD:
+            retVal->ChangeType(nativeReturnType);
+            break;
+
 #if defined(FEATURE_SIMD) || defined(FEATURE_HW_INTRINSICS)
 #ifdef FEATURE_SIMD
         case GT_SIMD:
@@ -3705,16 +3709,6 @@ void Lowering::LowerRetStruct(GenTreeUnOp* ret)
         }
         break;
 #endif // FEATURE_SIMD || FEATURE_HW_INTRINSICS
-
-        case GT_LCL_FLD:
-        {
-#ifdef DEBUG
-            LclVarDsc* varDsc = comp->lvaGetDesc(retVal->AsLclFld());
-            assert(varDsc->lvDoNotEnregister);
-#endif
-            retVal->ChangeType(nativeReturnType);
-        }
-        break;
 
         default:
             assert(varTypeIsEnregisterable(retVal));

